src/main/java/appeng/util/item/ItemList.java

```diff
@@ -219,13 +216,13 @@ public void resetStatus()
 
 	private IAEItemStack putItemRecord( final IAEItemStack itemStack )
 	{
-		return this.records.put( itemStack, itemStack );
+		return this.records.put( ( (AEItemStack) itemStack ).getSharedStack(), itemStack );
 	}
 
 	private Collection<IAEItemStack> findFuzzyDamage( final AEItemStack filter, final FuzzyMode fuzzy, final boolean ignoreMeta )
 	{
-		final IAEItemStack low = filter.getLow( fuzzy, ignoreMeta );
-		final IAEItemStack high = filter.getHigh( fuzzy, ignoreMeta );
+		final AESharedItemStack low = ( (AEItemStack) filter ).getSharedStack().getLow( fuzzy, ignoreMeta );
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "AEItemStack". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1905)